include/zephyr/dt-bindings/adc/stm32f4_adc.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2023 STMicrelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_STM32F4_ADC_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_STM32F4_ADC_H_
+
+#include <zephyr/dt-bindings/adc/stm32_adc.h>
+
+#define CFGR_REG	0x04
+#define RES_SHIFT	24
+#define RES_MASK	0x03
+
+#define STM32_ADC_RES(resolution, reg_val)	\
+	STM32_ADC(resolution, reg_val, RES_MASK, RES_SHIFT, CFGR_REG)
```

Comment:
This macro could be moved into `stm32_adc.h` and not duplicated in each chip specific header.  The only difference would be on stm32f1, where `STM32_ADC_RES(12)` would become `STM32_ADC_RES(12, 0x00);`.  Maybe that's not quite as good for f1?  But it's a nice refactor otherwise and only a small change on the f1.

The H7 header would still have the special adc3 version.

Reply:
Done